### PR TITLE
[MOB-5386] create flex message data type

### PR DESCRIPTION
**Pull request by roninopf**
## 🔹 Jira Ticket(s)

* [MOB-5386](https://iterable.atlassian.net/browse/MOB-5386)

## ✏️ Description

> Please provide a brief description of what this pull request does.

[MOB-5386]: https://iterable.atlassian.net/browse/MOB-5386?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: swift-sdk/IterableFlexMessage.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        let metadata = FlexMessageMetadata(id: id)
+        
+        self.init(metadata: metadata)
+    }
+}
+
+extension IterableFlexMessage {
+    struct FlexMessageMetadata {
+        let id: String
+        let placementId: String?
```
Comment (evangreer91) [inline swift-sdk/IterableFlexMessage.swift:33]: placementId should be required.

File: swift-sdk/IterableFlexMessage.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    
+    struct FlexMessageElements {
+        struct FlexMessageElementsButton {
+            let id: String
+            let title: String?
+            let action: String?
+        }
+        
+        struct FlexMessageElementsImages {
+            let id: String
```
Comment (evangreer91) [inline swift-sdk/IterableFlexMessage.swift:53]: This should be FlexMessageElementsImage to follow the convention.

**Review by evangreer91 — COMMENTED** [review]
Added a few comments. Good for my understanding to look through this.

File: swift-sdk/IterableFlexMessage.swift [hunk 3]
```diff
@@ -0,0 +1,67 @@
+//
+//  Copyright © 2022 Iterable. All rights reserved.
+//
+
+import Foundation
+
+public struct IterableFlexMessage {
+    let metadata: FlexMessageMetadata
+    let elements: FlexMessageElements?
+    let custom: [AnyHashable: Any]?
```
[the remark this answers is not in the document]
Comment (roninopf) [reply swift-sdk/IterableFlexMessage.swift:10]: Not sure what you mean, it should be just that.

File: swift-sdk/IterableFlexMessage.swift (hunk 1 above)
Comment (roninopf) [reply swift-sdk/IterableFlexMessage.swift:33]: Good catch!

File: swift-sdk/IterableFlexMessage.swift (hunk 2 above)
Comment (roninopf) [reply swift-sdk/IterableFlexMessage.swift:53]: Good catch here too!

File: swift-sdk/IterableFlexMessage.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ayyanchira) [reply swift-sdk/IterableFlexMessage.swift:10]: Ah I think Evan is talking about line 11. where there is another `payload` variable apart from `custom` too.

[the remark this answers is not in the document]
Comment (evangreer91) [reply swift-sdk/IterableFlexMessage.swift:10]: Yes shouldn't there be only one custom payload?

[the remark this answers is not in the document]
Comment (roninopf) [reply swift-sdk/IterableFlexMessage.swift:10]: Oh, that, I think I'll remove it for now. The intent was that it was the JSON payload (in its entirety), but we shouldn't bother with that (if we want it) until later anyways.

**Review by evangreer91 — APPROVED** [review]
LGTM :)

**Pull request merged**